modules/order/commerce_order.module

```diff
@@ -80,6 +80,13 @@ function template_preprocess_commerce_order(&$variables) {
   foreach (Element::children($variables['elements']) as $key) {
     $variables['order'][$key] = $variables['elements'][$key];
   }
+
+  if ($order->getBillingProfile()) {
+    $variables['order']['billing_information']['address'] = $order->getBillingProfile()->get('address')->view([
+      'type' => 'address_default',
+      'label' => 'hidden',
```

Comment:
Same comment as on the admin order view, rendering the address instead of the whole billing profile means that any tax info / phone fields won't be shown automatically, which is probably undesirable.